py/src/braintrust/test_logger.py

```diff
@@ -1994,11 +1994,13 @@ def inner():
 def reset_id_generator_state():
     """Reset ID generator state and environment variables before each test"""
     logger._state._reset_id_generator()
+    logger._state._reset_context_manager()
```

Comment:
we should start nuking the state between tests, but we can do that in the future.